### PR TITLE
Fix/eos submission

### PR DESCRIPTION
**Pull request by jglaser**
This fixes two issues when trying to submit MPI jobs on EOS.

1. EOS uses a qsub wrapper (a perl script), which expects to run inside a tty.
2. the # of ranks per node was wrong (it is 32, not 24)

## Description
ad 1. to honor the shebang (#1) line for perl of the qsub wrapper, apparently only subprocess.Popen works, but check_output does not. To additionally create a tty, one has to use python's pty lib.

## Motivation and Context
Trying to submit on OLCF eos.

## Types of Changes

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

This pull request is based on
- [x] *master*, because it is a bug fix or an update to the documentation.
- [ ] *develop*, because it introduces a new feature.

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.txt).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/scheduling/torque.py [hunk 1]
```diff
@@ -141,7 +141,11 @@ def submit(self, script, after=None, pretend=False, hold=False, flags=None, *arg
     def is_present(cls):
         "Return True if it appears that a TORQUE scheduler is available within the environment."
         try:
-            subprocess.check_output(['qsub', '--version'], stderr=subprocess.STDOUT)
+            # some qsub wrappers don't run on no-tty
+            import pty
+            master, slave = pty.openpty()
+            subprocess.Popen(['qsub', '--version'], stdin=slave, stdout=slave, stderr=slave,
+                             shell=True)
```
Comment (csadorf) [inline flow/scheduling/torque.py]: It appears that this function is not reliably raising either an `IOError` or `OSError` to indicate the scheduler is *not* present.

For example, I executed this command on a machine that I know has no Torque scheduling system: 
```
$ python -c "from flow.scheduling import TorqueScheduler; print(TorqueScheduler.is_present())"
True
```
I'm pretty sure that this is also the reason why unit tests are failing.

File: flow/templates/eos.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@
 {% if operations|calc_tasks('ngpu', true, true) and not force %}
 {% raise "GPUs were requested but are unsupported by Eos!" %}
 {% endif %}
-{% set nn = nn|default(cpu_tasks|calc_num_nodes(24), true) %}
+{% set nn = nn|default(cpu_tasks|calc_num_nodes(32), true) %}
```
Comment (csadorf) [inline flow/templates/eos.sh]: Thank you for catching this.

**Comment by csadorf** [comment]
@jglaser I've scheduled this PR for the v0.7, however I've identified some issues with the patch which would need to be resolved prior to merging.

Do you need any assistance with this? Maybe we just split this so that the problematic changes to the `torque.py` module can be addressed separately?

**Comment by jglaser** [comment]
> @jglaser I've scheduled this PR for the v0.7, however I've identified some issues with the patch which would need to be resolved prior to merging.
> 
> Do you need any assistance with this? Maybe we just split this so that the problematic changes to the `torque.py` module can be addressed separately?

I will try to get to it soon, perhaps over the weekend. Feel free to fix the 24->32 issue separately otherwise.

**Comment by csadorf** [comment]
> > @jglaser I've scheduled this PR for the v0.7, however I've identified some issues with the patch which would need to be resolved prior to merging.
> > Do you need any assistance with this? Maybe we just split this so that the problematic changes to the `torque.py` module can be addressed separately?
> 
> I will try to get to it soon, perhaps over the weekend. Feel free to fix the 24->32 issue separately otherwise.

Ok, sounds good. I'll move this to v0.8 and fix the 24->32 issue as part of #69 .

**Comment by csadorf** [comment]
Sorry, I got confused with the dates, there is still more than enough time to make this part of v0.7.

**Comment by csadorf** [comment]
@jglaser I'm planning on releasing 0.7.0 either today or tomorrow? Should we reschedule this for 0.8.0?

**Comment by jglaser** [comment]
For some reason I *was* able to submit on eos with the new `feature/enable-aggregate-operations` branch, which includes develop. Perhaps this branch is now superfluous?

**Comment by csadorf** [comment]
> For some reason I _was_ able to submit on eos with the new `feature/enable-aggregate-operations` branch, which includes develop. Perhaps this branch is now superfluous?

It might be. I can leave it open a bit longer if you like. Otherwise I'll decline it.

**Comment by csadorf** [comment]
I'll close this for now.

**Pull request closed without merging**